File: _src/03/Chapter3.VB/PersonMap.vb
Imports System.Data.Entity.ModelConfiguration

Public Class PersonMap
    Inherits EntityTypeConfiguration(Of Person)

    Public Sub New()
        Me.Property(Function(p) p.FirstName) _
           .HasMaxLength(30) _
           .IsRequired()
        Me.Property(Function(p) p.LastName) _
            .HasMaxLength(30) _ 
            .IsRequired()
        Me.Property(Function(p) p.MiddleName) _
            .HasMaxLength(1).IsFixedLength().IsUnicode(False)
        Me.Property(Function(p) p.BirthDate) _
            .HasPrecision(1)
        Me.Property(Function(p) p.HeightInFeet) _
           .HasPrecision(4, 2)
        Me.Property(Function(p) p.Photo) _
            .IsVariableLength().HasMaxLength(4000)
    End Sub

End Class

File: _src/03/Chapter3.CSharp/PersonMap.cs
using System.Data.Entity.ModelConfiguration;

namespace Chapter3.CSharp
{
    public class PersonMap : EntityTypeConfiguration<Person>
    {
        public PersonMap()
        {
            Property(p => p.FirstName)
                .HasMaxLength(30)
                .IsRequired();
            Property(p => p.LastName)
                .IsRequired()
                .HasMaxLength(30);
            Property(p => p.MiddleName)
                .HasMaxLength(1)
                .IsFixedLength()
                .IsUnicode(false);
            Property(p => p.BirthDate)
                .HasPrecision(10);
            Property(p => p.HeightInFeet)
                .HasPrecision(4, 2);
            Property(p => p.Photo)
                .IsVariableLength()
                .HasMaxLength(4000);
            HasMany(p => p.Phones)
                .WithRequired()
                .HasForeignKey(ph => ph.PersonId);
        }
    }
}
